### PR TITLE
Implement mosaic reveal block

### LINKED ISSUE #74: Implement Mosaic reveal block
Original: https://www.esri.com/en-us/about/about-esri/mea-central-asia
New: https://main--esri--aemsites.hlx.page/en-us/about/about-esri/mea-central-asia

image

### PR DESCRIPTION
**Pull request by vladciocoiu**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #74 

Test URLs:
- Original: https://www.esri.com/en-us/about/about-esri/mea-central-asia
- Before: https://main--esri--aemsites.hlx.live/drafts/vciocoiu/mosaic-reveal
- After: https://add-mosaic-reveal--esri--aemsites.hlx.live/drafts/vciocoiu/mosaic-reveal

### CODE REVIEW

File: blocks/mosaic-reveal/mosaic-reveal.css [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+@media (width >= 1450px) {
+  .mosaic-reveal {
+    .title {
+      padding: var(--space-3) var(--space-12) var(--space-12) var(--space-12);
+    }
+  }
+}
+
+#globalfooter {
```
Comment (alexcarol) [inline blocks/mosaic-reveal/mosaic-reveal.css:215]: is this just for debugging?

**Comment by alexcarol** [comment]
Regarding the animation, on the live site the element appears to be moving ("translateY") from the bottom, rather than scaling in the current implementation. Could you give a short to that approach?

**Comment by vladciocoiu** [comment]
> Regarding the animation, on the live site the element appears to be moving ("translateY") from the bottom, rather than scaling in the current implementation. Could you give a short to that approach?

I animated the inset-block-start, since that was the approach in the live site. Thanks for pointing it out!

File: blocks/mosaic-reveal/mosaic-reveal.css [hunk 2]
```diff
@@ -0,0 +1,215 @@
+.mosaic-reveal-container {
+  display: flex;
```
Comment (alexcarol) [inline blocks/mosaic-reveal/mosaic-reveal.css:2]: We don't want to couple the mosaic-reveal to this behaviour, can you implement it with the selector"column-section" (https://github.com/aemsites/esri/issues/73) instead? (it's added via column metadata, you can put the code in styles.css)

Comment (vladciocoiu) [reply blocks/mosaic-reveal/mosaic-reveal.css:2]: So I should add
```
  display: flex;
  justify-content: center;
  align-items: center;
```
to `.column-section`?

File: blocks/mosaic-reveal/mosaic-reveal.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    buttonContainer.parentElement.appendChild(link);
+    buttonContainer.parentElement.removeChild(buttonContainer);
+  });
+
+  const revealContent = div({ class: 'mosaic-reveal-content', 'aria-hidden': 'true' });
+  revealContent.addEventListener('click', () => {
+    revealContent.setAttribute('aria-hidden', 'true');
+  });
+
+  window.addEventListener('resize', () => {
```
Comment (alexcarol) [inline blocks/mosaic-reveal/mosaic-reveal.js:37]: you might want to use window.matchMedia instead:
https://github.com/hlxsites/moleculardevices/blob/15619942358776750c1f4280730f98b6e4e6cd37/blocks/citation-list/citation-list.js#L71
https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia

File: blocks/mosaic-reveal/mosaic-reveal.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    child.addEventListener('click', () => {
+      if (window.innerWidth < 860) {
+        revealContent.setAttribute('aria-hidden', 'false');
+        const children = [...contents[idx].children].map((el) => el.cloneNode(true));
+        revealContent.replaceChildren(...children);
+      }
+    });
+
+    child.addEventListener('mouseenter', () => {
+      if (window.innerWidth >= 860) {
```
Comment (alexcarol) [inline blocks/mosaic-reveal/mosaic-reveal.js:57]: same here for matchMedia

**Review by alexcarol — APPROVED** [review]

**Pull request merged**